genesyscloud/proxies/architect_api/architect_ivr.go

```diff
@@ -29,36 +28,90 @@ type ArchitectIvrProxy struct {
+	putArchitectIvrBasic putArchitectIvrFunc
 }
 
 func NewArchitectIvrProxy() *ArchitectIvrProxy {
 	var architectApi *platformclientv2.ArchitectApi
 	return &ArchitectIvrProxy{
 		Api: architectApi,
-		PostArchitectIvr: func(api *platformclientv2.ArchitectApi, ivr *platformclientv2.Ivr) (*platformclientv2.Ivr, *platformclientv2.APIResponse, error) {
-			return api.PostArchitectIvrs(*ivr)
-		},
-		GetArchitectIvr: func(api *platformclientv2.ArchitectApi, id string) (*platformclientv2.Ivr, *platformclientv2.APIResponse, error) {
-			return api.GetArchitectIvr(id)
-		},
-		PutArchitectIvr: func(api *platformclientv2.ArchitectApi, id string, ivr *platformclientv2.Ivr) (*platformclientv2.Ivr, *platformclientv2.APIResponse, error) {
-			return api.PutArchitectIvr(id, *ivr)
-		},
-		DeleteArchitectIvr: func(api *platformclientv2.ArchitectApi, id string) (*platformclientv2.APIResponse, error) {
-			return api.DeleteArchitectIvr(id)
-		},
-		GetArchitectIvrs: func(api *platformclientv2.ArchitectApi, pageNumber int, pageSize int, sortBy, sortOrder, name, dnis, scheduleGroup string) (*platformclientv2.Ivrentitylisting, *platformclientv2.APIResponse, error) {
-			return api.GetArchitectIvrs(pageNumber, pageSize, sortBy, sortOrder, name, dnis, scheduleGroup)
-		},
-		UploadIvrDnisChunks: uploadIvrDnisChunks,
+
+		PostArchitectIvr:   postArchitectIvr,
+		GetArchitectIvr:    getArchitectIvr,
+		PutArchitectIvr:    putArchitectIvr,
+		DeleteArchitectIvr: deleteArchitectIvr,
+		GetArchitectIvrs:   getArchitectIvrs,
+
+		MaxDnisPerRequest: 50,
+
+		putArchitectIvrBasic: putArchitectIvrBasic,
 	}
 }
 
 func (a *ArchitectIvrProxy) ConfigureProxyApiInstance(c *platformclientv2.Configuration) {
 	a.Api = platformclientv2.NewArchitectApiWithConfig(c)
 }
 
+func postArchitectIvr(a *ArchitectIvrProxy, ivr *platformclientv2.Ivr) (*platformclientv2.Ivr, *platformclientv2.APIResponse, error) {
+	dnis := *ivr.Dnis
+
+	dnisChunks := utillists.ChunkStringSlice(dnis, a.MaxDnisPerRequest)
+	if len(dnisChunks) == 1 {
+		ivr.Dnis = &dnisChunks[0]
+	}
+
+	respIvr, resp, err := a.Api.PostArchitectIvrs(*ivr)
+
+	if len(dnisChunks) > 1 {
+		respIvr, _, err := uploadIvrDnisChunks(a, dnisChunks, *respIvr.Id)
+		if err != nil {
+			return respIvr, nil, err
+		}
+	}
+
+	return respIvr, resp, err
+}
+
+func getArchitectIvr(api *platformclientv2.ArchitectApi, id string) (*platformclientv2.Ivr, *platformclientv2.APIResponse, error) {
+	return api.GetArchitectIvr(id)
+}
+
+func putArchitectIvr(a *ArchitectIvrProxy, id string, ivr *platformclientv2.Ivr) (*platformclientv2.Ivr, *platformclientv2.APIResponse, error) {
+	var dnisChunks [][]string
+	if ivr.Dnis != nil {
+		dnisChunks = utillists.ChunkStringSlice(*ivr.Dnis, a.MaxDnisPerRequest)
+		if len(dnisChunks) != 1 {
+			ivr.Dnis = nil
+		}
+	}
+
+	respIvr, resp, err := a.putArchitectIvrBasic(a, id, ivr)
+
+	if len(dnisChunks) > 1 {
+		respIvr, _, err = uploadIvrDnisChunks(a, dnisChunks, id)
```

Comment:
Just curious as to why you did not make the uploadIvrDnsChunks as a private method on the structure.  My main concern with the previous rework is having the uploadIvrDnsChunks being called from outside the struct when the code that relied on it was in a struct.

Reply:
I was thinking if this function is only needed internally within the package, it wouldn't belong on the struct. And I assumed the method name starting with a lower case letter would make it private
Are you suggesting defining the method as `func (a  *ArchitectIvrProxy) uploadIvrDnisChunks(...` rather than pasing the struct object in as a param. Thinking about it now, that would look better here

Reply:
@carnellj-genesys Ah, I think I needed more time to sit with this. I should have kept the uploadChunk methods as part of the struct so I can mock them out and test the outer function (attached to the public field) that calls them